Feat(Insomnia): How Insomnia ships (#5125)
Kong/developer.konghq.com: app/insomnia/release-policy.md

Reviewer: rename the file

  - /insomnia/

products:
  - insomnia

search_aliases:
  - updates
  - releases

description: How Insomnia releases new versions, with recommendations for updating.

related_resources:
  - text: Stages of software availability
    url: /stages-of-software-availability/

---

## Overview

Insomnia uses [semantic versioning](https://semver.org/), a three-part number format: MAJOR.MINOR.PATCH (for example `2.5.1`) for defining versions. 

Insomnia communicates all changes via:

- A [changelog](https://insomnia.rest/changelog).

- The [GitHub releases page](https://github.com/kong/insomnia/releases).

Public documentation updates support the latest versions of Insomnia.

## Release types

- **Generally Available (GA)**: Released to all users.
- **Beta**: A 7-day period before GA when users can opt in to test the release and provide early feedback. Opt in via **Settings > Release Channel** in the Insomnia app.
- **Preview** or **Tech Preview**: Features that are ready to use but still in active development. Preview features are clearly marked in the product.

Learn more about stages of software availability [on the dedicated page](/stages-of-software-availability/).
{:.info}

> > Insomnia strongly recommends using the newest version, and that all users collaborating on the same project remain on the same major version.

Author: ```suggestion

{:.info}
> Insomnia strongly recommends using the newest version, and that all users collaborating on the same project remain on the same major version.
```

## Patch version changes

Insomnia ships a patch version when a release contains only backwards-compatible bug fixes, corrections that make Insomnia behave the way it was already documented or intended to behave.

Examples include:

- Fixing a crash or hang during request execution.
- Correcting response rendering or syntax highlighting errors.
- Fixing import or export errors that prevented valid collections from loading.
- Fixing authentication flows that didn't complete correctly.
- Fixing environment variable resolution or template tag evaluation.
- Correcting UI rendering glitches or layout issues.
- Resolving sync or collaboration conflicts, including data loss scenarios.
- Patching security vulnerabilities in bundled dependencies, with no user-facing behavior change.
- Fixing incorrect HTTP header handling or encoding.
- Correcting documentation links or in-app help text.

## Send feedback

Users can provide feedback depending on the type of plan they are on:

{% feature_table %}
item_title: Feedback
columns:
  - title: Essential
    key: essential
  - title: Pro
    key: pro
  - title: Enterprise
    key: enterprise

features:

>   - title: GitHub issues for defects or ideas.

Reviewer: ```suggestion
  - title: GitHub issues for bug reports and feature requests
```

Reviewer: Ideas and defects feels too casual.

---
title: Insomnia release and versioning policy
content_type: reference
layout: reference
breadcrumbs:
  - /insomnia/

products:
  - insomnia

search_aliases:
  - updates
  - releases

description: How Insomnia releases new versions, with recommendations for updating.

related_resources:
  - text: Stages of software availability
    url: /stages-of-software-availability/

---

## Overview

Insomnia uses [semantic versioning](https://semver.org/), a three-part number format: MAJOR.MINOR.PATCH (for example `2.5.1`) for defining versions. 

Insomnia communicates all changes via:

- A [changelog](https://insomnia.rest/changelog).

- The [GitHub releases page](https://github.com/kong/insomnia/releases).

Public documentation updates support the latest versions of Insomnia.

## Release types

- **Generally Available (GA)**: Released to all users.
- **Beta**: A 7-day period before GA when users can opt in to test the release and provide early feedback. Opt in via **Settings > Release Channel** in the Insomnia app.
- **Preview** or **Tech Preview**: Features that are ready to use but still in active development. Preview features are clearly marked in the product.

> 

Author: ```suggestion

Learn more about stages of software availability [on the dedicated page](/stages-of-software-availability/).
```

> ### Recommendations

Reviewer: ```suggestion
{:.info}
```

> Insomnia strongly recommends using the newest version, and that all users collaborating on the same project remain on the same major version.

Reviewer: ```suggestion
> Insomnia strongly recommends using the newest version, and that all users collaborating on the same project remain on the same major version.
```

## Long Term Support (LTS)

Insomnia doesn't have an explicit Long Term Support (LTS) policy, nor does it publish LTS versions.

## Backports

> Fixes and changes don't backport to earlier versions. Best practice is to upgrade to the latest version.

Reviewer: ```suggestion
Fixes and changes don't backport to earlier versions. The best practice is to upgrade to the latest version.
```

> Insomnia doesn't roll new or modified features into prior versions except in extraordinary circumstances, at our sole discretion.

Reviewer: ```suggestion
```
I think this is repeat information

## Types of Versions

{% table %}
columns:
  - title: Version
    key: version
  - title: When
    key: time
  - title: Content
    key: content
  - title: Recommendations
    key: recommendation
rows:
  - version: "Major (ex: v12)"
    time: |
      - Twice a year
      - Beta for 7 days before GA
    content: |
      - New features
      - Bug fixes
      - Architecture improvements
      - Changes or fixes that aren't backwards compatible.
    recommendation: |
      - Test the beta first to ensure compatibility.
      - Backup your work before updating.
  - version: "Minor (ex: v12.1)"
    time: "Approximately every three weeks"
    content: "Changes that are always compatible across its parent major version"
    recommendation: "Usually automated migrations, but manual migrations may be necessary on some releases."
  - version: "Patch (ex: v12.1.1)"
    time: "When needed on a minor version"
    content: |
      Only:

      - Defect fixes
      - Security patches
      - Other maintenance work
{% endtable %}

## Major version changes

> Insomnia ships a major version when the release contains breaking changes. Breaking changes count as anything that, for the majority of users, could cause an existing workflow, configuration, or integration to stop working after the upgrade.

Reviewer: ```suggestion
Insomnia ships a major version when the release contains breaking changes. Breaking changes count as anything that could cause an existing workflow, configuration, or integration to stop working after the upgrade.
```
Better to remove the qualifier, if it breaks workflows its probably a major version ship

For example:

- Removing a request type, protocol, or authentication method.
- Removing or renaming a CLI command, flag, or subcommand.
- Changing the export schema for collections or environments in a way that older versions can't read.
- Dropping support for an operating system, platform, or runtime.
- Changing the plugin API in a way that breaks existing plugins.
- Removing or renaming a public API endpoint.
- Changing the default behavior of a feature in a way that affects existing users.

> - Migrating the underlying data store to a format that can't be rolled back to a prior version.

Reviewer: ```suggestion
- Migrating the underlying data store to a format that can't be incompatible with prior versions.
```

- Adding new environment variable types or template tag functions.
- Adding a new integration, such as a Git provider or CI/CD tool.
- Adding new configuration options or preferences.

## Patch version changes

Insomnia ships a patch version when a release contains only backwards-compatible bug fixes, corrections that make Insomnia behave the way it was already documented or intended to behave.

Examples include:

- Fixing a crash or hang during request execution.
- Correcting response rendering or syntax highlighting errors.
- Fixing import or export errors that prevented valid collections from loading.
- Fixing authentication flows that didn't complete correctly.
- Fixing environment variable resolution or template tag evaluation.
- Correcting UI rendering glitches or layout issues.
- Resolving sync or collaboration conflicts, including data loss scenarios.
- Patching security vulnerabilities in bundled dependencies, with no user-facing behavior change.
- Fixing incorrect HTTP header handling or encoding.
- Correcting documentation links or in-app help text.

## Types of Releases

- Generally Available (GA): Available to anyone.

> - Beta: Generally a 7\-day period before GA where clients can opt in (via Settings -> Release Channel in the app) and provide early [feedback](#user-feedback).

Author: ```suggestion

```

## Send feedback

Users can provide feedback depending on the type of plan they are on:

{% feature_table %}
item_title: Feedback
columns:
  - title: Essential
    key: essential
  - title: Pro
    key: pro
  - title: Enterprise
    key: enterprise

features:
  - title: GitHub issues for defects or ideas.
    essential: true
    pro: true
    enterprise: true
  - title: Personalized support through support@insomnia.rest
    essential: false
    pro: true
    enterprise: true

>   - title: Sending feedback to a dedicaded Customer Success Manager (CSM).

Author: ```suggestion
  - title: Sending feedback to a dedicated Customer Success Manager (CSM).
```

---
title: Insomnia release and versioning policy
content_type: reference
layout: reference
breadcrumbs:
  - /insomnia/

products:
  - insomnia

search_aliases:
  - updates
  - releases

description: How Insomnia releases new versions, with recommendations for updating.

related_resources:
  - text: Stages of software availability
    url: /stages-of-software-availability/

---

## Overview

Insomnia uses [semantic versioning](https://semver.org/), a three-part number format: MAJOR.MINOR.PATCH (for example `2.5.1`) for defining versions. 

Insomnia communicates all changes via:

- A [changelog](https://insomnia.rest/changelog).

- The [GitHub releases page](https://github.com/kong/insomnia/releases).

Public documentation updates support the latest versions of Insomnia.

> 

Reviewer: ```suggestion
## Release types

- **Generally Available (GA)**: Released to all users.
- **Beta**: A 7-day period before GA when users can opt in to test the release and provide early feedback. Opt in via **Settings > Release Channel** in the Insomnia app.
- **Preview** or **Tech Preview**: Features that are ready to use but still in active development. Preview features are clearly marked in the product.
```

I moved this higher up in the document and tried to simplify the language.

- Deprecating (but not yet removing) a feature, command, or API.
- Adding new environment variable types or template tag functions.
- Introducing a new integration (for example a new Git provider, a new CI/CD integration).
- Adding new configuration options or preferences.

## Changes in Patch versions

Insomnia ships a patch version when the release contains only backwards-compatible bug fixes (corrections that make Insomnia behave the way it was already documented or intended to behave).

For example:

- Fixing a crash or hang during request execution.
- Correcting response rendering or syntax highlighting errors.
- Fixing import/export issues where valid collections failed to load.
- Resolving authentication flows that were not completing correctly.
- Fixing environment variable resolution or template tag evaluation bugs.
- Correcting UI rendering glitches or layout issues.
- Fixing sync/collaboration conflicts or data loss scenarios.
- Updating bundled dependencies to patch security vulnerabilities (with no user-facing behavior change).
- Fixing incorrect HTTP header handling or encoding issues.
- Correcting documentation links or in-app help text. 

> ## Types of Releases

Reviewer: This should go higher up in the document I think.
